src/main/scala/edu/berkeley/ce/rockslicing/ClassRegistrator.scala

```diff
@@ -0,0 +1,24 @@
+package edu.berkeley.ce.rockslicing
+
+import com.esotericsoftware.kryo.io.{Input, Output}
+import com.esotericsoftware.kryo.{Kryo, Serializer}
+import com.twitter.chill.SingletonSerializer
+import org.apache.spark.serializer.KryoRegistrator
+
+/**
+  * Kryo serializer registrator for all classes that are to be serialized.
+  */
+class ClassRegistrator extends KryoRegistrator {
+  override def registerClasses(kryo: Kryo) {
+    kryo.register(None.getClass, new SingletonSerializer[AnyRef](None))
+    kryo.register(Nil.getClass, new SingletonSerializer[AnyRef](Nil))
+    kryo.register(Some.getClass, new SingletonSerializer[AnyRef](Some))
+    kryo.register(classOf[scala.collection.mutable.WrappedArray$ofRef])
```

Comment:
To be honest, I'm not sure. The stack trace doesn't give any information on where these `WrappedArray`s are coming from, it only tells you that it is not registered with Kryo.  The offending instance could be coming from any of the remaining `Seq`s in our code - it doesn't mean that the changes that were made to get rid of boxed `double`s isn't working. Do you agree with that or am I missing something?